### DISCUSSION TITLE
Error while registering custom web component with search results WP.

### DISCUSSION THREAD

**Discussion by GangadharMFord** [microsoft-search/pnp-modern-search#2787] [Q&A]
Hi Team,

I have a custom requirement to add a preference language dropdown at search results webpart.  To achieve this were i have developed my custom component at pnp-modern-search-extensibility-demo solution and configured as per the below supporting article but still i couldn't see the custom component into the search results webpart. Am I missed any configuration?

Followed link: https://microsoft-search.github.io/pnp-modern-search/extensibility/custom_web_component/

please suggest?

And if I add this custom component into the available components section with Search-parts core madule then the below error is coming.
Error:
ERROR:
Failed to execute 'define' on 'CustomElementRegistry': "pnp-preferenceLanguage" is not a valid custom element name

CALL STACK:
Error: Failed to execute 'define' on 'CustomElementRegistry': "pnp-preferenceLanguage" is not a valid custom element name
    at https://localhost:4321/dist/modern-search-results-web-part.js:79881:51
    at Array.forEach (<anonymous>)
    at https://localhost:4321/dist/modern-search-results-web-part.js:79868:35
    at e.whenFinished (https://res-1.cdn.office.net/files/sp-client/sp-webpart-workbench-assembly_en-us_e42c39e38f9d9b7799831f1f767d2daa.js:70:535725)
    at https://localhost:4321/dist/modern-search-results-web-part.js:79866:35
    at new Promise (<anonymous>)
    at TemplateService.<anonymous> (https://localhost:4321/dist/modern-search-results-web-part.js:79859:20)
    at Generator.next (<anonymous>)
    at https://localhost:4321/dist/modern-search-results-web-part.js:79645:71
    at new Promise (<anonymous>)

Please suggest the details steps to configure any custom web component to the pnp modern search solution.
thank you.

**Comment by wobba** [comment]
Did you build it with the same SPFx version as the webparts?

**Reply by GangadharMFord** [reply]
Yes, the SPFx version for both solutions will be (search-parts and pnp-modern-search-extensibility-demo solution) are same versions.
but still have the issue as couldn't able to see the custom web component into search results webpart.

if possible, can you share brief steps that we need to follow,

**Reply by wobba** [reply]
Did you do this step? https://microsoft-search.github.io/pnp-modern-search/extensibility/#register-your-extensibility-library-with-a-web-part

**Reply by GangadharMFord** [reply]
Hi, Yes. I completly followed the steps which is mentioned in the above article and able to see and render the default custom component into the search results webpart but other custom component (basically its dropdown with values) are not coming.. 

I have checked all the configurations and mappings for default custom component with my new components and all same but still not accessable from search results webpart.

For testing I did added a text/ dropdown on custom component as well but still the changes to the default component is not reflecting int o the results,

Any suggestions?

**Reply by wobba** [reply]
Not sure how best to help you troubleshoot this. Run the web part in debug mode and see where it loads etc maybe?

**Reply by GangadharMFord** [reply]
> Not sure how best to help you troubleshoot this. Run the web part in debug mode and see where it loads etc maybe?

Hi Wobba, thank you always for your help and would like to cross check with you regarding both solution build versions.
currently my main solution (i.e. search parts) solution is 4.7 and modern Search Extensibility sample is "version": "4.1.0.0",

but still I can't using this solution with search parts ? could you please help me on which version [Modern Search Extensibility will support with 4.7 version?

**Reply by wobba** [reply]
You need to re-build extensibility to match the SPFx version of the web parts. For 4.7 that is 1.14.0.

**Reply by GangadharMFord** [reply]
> You need to re-build extensibility to match the SPFx version of the web parts. For 4.7 that is 1.14.0.

So I just quickly checked the SPFx version for main (search parts) solution and extensibility  solution and here is the findings. here is the snpat shot.
I ran the below command for get SPFx version details and here is the complete details. please guide me if anything wrong!!!!
npm ls -g --depth=0 @microsoft/generator-sharepoint

![image](https://user-images.githubusercontent.com/111119555/224913866-ea24da82-a2ea-485f-afb9-98e781981025.png)

So the below command will help me on setting SPFx version to 1.14.0 ?
**npm install gulp-cli yo @microsoft/generator-sharepoint@1.14.0 --global**

**Reply by wobba** [reply]
You can use `yo @microsoft/sharepoint --skip-install`, then edit package json to specify the correct version, then run `npm i`. That is one way and you should be able to keep the latest yeoman version. Or install the older one.

**Reply by GangadharMFord** [reply]
> You can use `yo @microsoft/sharepoint --skip-install`, then edit package json to specify the correct version, then run `npm i`. That is one way and you should be able to keep the latest yeoman version. Or install the older one.

I followed the same and still have the same issue and couldn't able to integrate extensibility solution with Main solution.
but i could able to see out of the custom suggestions at search box not if we add new custom suggestion other than out of the box...

could you please share me any standard steps we have anything ...

here are the current configurations which search parts and extensibility solutions.

PS C:\MyProject\pnp-modern-search\search-parts> npm ls -g --depth=0
C:\Program Files\nodejs
+-- @microsoft/generator-sharepoint@1.14.0
+-- @pnp/modern-search-extensibility@1.4.0
+-- gulp-cli@2.3.0
+-- npm@6.14.17
+-- tslint@6.1.3
+-- typescript@4.8.4

PS C:\MyProject\pnp-modern-search-extensibility-samples\search-extensibility-samples> npm ls -g --depth=0
C:\Program Files\nodejs
+-- @microsoft/generator-sharepoint@1.14.0
+-- @pnp/modern-search-extensibility@1.4.0
+-- gulp-cli@2.3.0
+-- npm@6.14.17
+-- tslint@6.1.3
+-- typescript@4.8.4
`-- yo@4.3.1


Very quickly i created another SharePoint site and deployed 4.8 solution with followed by 1.12.0 extensibility sample into the site (site collection app cateloge - site scope) were tested the custom suggestions and its working (its showing all new custom suggestion entried into the search box) but where I add a custom web component still the same same with error.
TECHNICAL DETAILS
ERROR:
Failed to execute 'define' on 'CustomElementRegistry': "Custom-PreferenceLanguageComponent" is not a valid custom element name

**Reply by GangadharMFord** [reply]
@wobba , could you please help on this issue with your suggestions and integration part? thank you.

**Reply by GangadharMFord** [reply]
@wobba @FranckyC , As part of fixing the Spfx version issue, I have complete deleted the node modules and executed for 1.14.0 version command below. post that i tested at workbench and search results webpart no loading. here is the snapshot.. please help me on this.

![image](https://user-images.githubusercontent.com/111119555/227113534-c8cb86df-d032-4874-b41d-3a783e803cab.png)


But from search box, I could able to get custom suggestions from extensibility solution.
please help on the above error.
